### PR TITLE
Feat change to tekton output

### LINKED ISSUE #5: Task output size > 4k fails the task
**Describe the bug**
In case the Box task output param's content is > 4k, because of the existing Tekton underlying limitations, the task execution fails due to the size of the output. 

**To Reproduce**
Run a workflow which contains a box task with an output > 4k. 

**Expected behavior**
Introduce, where is needed, the path input field to store the content of the response param when it is used by the end-user.

**Environment (please complete the following information):**

- OS: any
- Browser any
- Version any

**Additional context**

### PR DESCRIPTION
**Pull request by cosmin-costache**
Closes #5 

#### Changelog

**New**

- added a new method to mask sensitive parameter values in logs. A default list of parameters is defined in EssentialsUtils class.

**Changed**

- added option --outputFilePath to 'list' command. If the additional parameter contains a valid file path, the task output will be written to this file
- added option --outputFilePath to 'info' command. Same as for the list command.
- updated log statements to use the new method for masking the parameter values.

### CODE REVIEW

**Comment by morarucostel** [comment]
Hey @cosmin-costache , thanks for the contribution.
Looks good to me.
Were you able to test it as part of a Task in a workflow?

**Review by morarucostel — APPROVED** [review]

**Comment by morarucostel** [comment]
@georgesafta can you please provide the +1 ?

File: src/main/java/com/ibm/garage/box/command/BoxCommand.java [hunk 1]
```diff
@@ -40,11 +41,21 @@ public CommandLine getCommandLine() {
...
-  public void list() {
+  public void list(
+        @Option(names = {"-o", "--outputFilePath"}, paramLabel = "<outputFilePath>")
+        String outputFilePath) {
     try {
       List<BoxFolderVo> folders = boxService.getFolders();
       ObjectMapper mapper = getObjectMapper();
-      output(map(STATUS, OK).add("folders", mapper.writeValueAsString(folders)));
+      String jsonResponse = mapper.writeValueAsString(folders);
+      if(outputFilePath != null && !outputFilePath.isEmpty()) {
```
Comment (georgesafta) [inline src/main/java/com/ibm/garage/box/command/BoxCommand.java:51]: can you please use the Apache `StringUtils.isNotBlank` method here?

File: src/main/java/com/ibm/garage/box/command/BoxCommand.java [hunk 2]
```diff
@@ -125,11 +136,21 @@ public void download(@Parameters(index = "0", paramLabel = "<fileId>") String fi
...
-  public void info(@Parameters(index = "0", paramLabel = "<folderId>") String folderId) {
+  public void info(@Parameters(index = "0", paramLabel = "<folderId>") String folderId,
+      @Option(names = {"-o", "--outputFilePath"}, paramLabel = "<outputFilePath>") 
+      String outputFilePath) {
     try {
       BoxFolderInfoVo folder = boxService.getFolder(folderId);
       ObjectMapper mapper = getObjectMapper();
-      output(map(STATUS, OK).add("folder", mapper.writeValueAsString(folder)));
+      String jsonResponse = mapper.writeValueAsString(folder);
+      if(outputFilePath != null && !outputFilePath.isEmpty()) {
```
Comment (georgesafta) [inline src/main/java/com/ibm/garage/box/command/BoxCommand.java:146]: can you please use the Apache `StringUtils.isNotBlank` method here?

File: src/main/java/com/ibm/garage/box/command/BoxCommand.java [hunk 3]
```diff
@@ -40,11 +41,21 @@ public CommandLine getCommandLine() {
...
+  public void list(
+        @Option(names = {"-o", "--outputFilePath"}, paramLabel = "<outputFilePath>")
+        String outputFilePath) {
     try {
       List<BoxFolderVo> folders = boxService.getFolders();
       ObjectMapper mapper = getObjectMapper();
-      output(map(STATUS, OK).add("folders", mapper.writeValueAsString(folders)));
+      String jsonResponse = mapper.writeValueAsString(folders);
+      if(outputFilePath != null && !outputFilePath.isEmpty()) {
+        LOGGER.info(String.format("Saving task output to file %s", outputFilePath));
```
Comment (georgesafta) [inline src/main/java/com/ibm/garage/box/command/BoxCommand.java:52]: no need for String.format here, you can use the SLF4J API directly here by:
```
LOGGER.info("Saving task output to file {}", outputFilePath);
```

**Review by georgesafta — CHANGES_REQUESTED** [review]
Also, can you please format the code using the Google formater from here : https://github.com/google/styleguide. Choose the java one suitable to your IDE.

**Review by georgesafta — APPROVED** [review]

**Comment by morarucostel** [comment]
Closing issue #5

**Pull request merged**